### ISSUE TITLE
status: slice bounds out of range panic in func wrap()

### ISSUE THREAD

**Issue by bazizsafi** [zrepl/zrepl#419]
We occasionally (every couple of month) get a `slice bounds out of range` error here...

https://github.com/zrepl/zrepl/blob/96d5288667f23bdda34327129e75967ab3ac0f28/client/status.go#L131

...when running `zrepl status` in a tmux session on our backup server.

It seems to be a "purely cosmetic" error. Restarting `zrepl status` just picks up again without seeming to have caused any problems with the snapshot creation/replication/pruning itself. So, just wanted to let you (and the community) know.

Here's the stacktrace:
```
panic: runtime error: slice bounds out of range [:-2]

goroutine 1 [running]:
github.com/zrepl/zrepl/client.wrap(0xc000113950, 0x43, 0xfffffffffffffffe, 0x2, 0x42)
        /src/client/status.go:131 +0x3c5
github.com/zrepl/zrepl/client.(*tui).printfDrawIndentedAndWrappedIfMultiline(0xc0000acff0, 0xcd47f9, 0xa, 0xc0003bf898, 0x1, 0x1)
        /src/client/status.go:150 +0x118
github.com/zrepl/zrepl/client.(*tui).renderPrunerReport(0xc0000acff0, 0xc0000acd70)
        /src/client/status.go:585 +0x9ae
github.com/zrepl/zrepl/client.(*tui).draw(0xc0000acff0)
        /src/client/status.go:345 +0x868
github.com/zrepl/zrepl/client.runStatus(0xdd2420, 0xc0001fed50, 0x121e440, 0x125d3b8, 0x0, 0x0, 0x0, 0x0)
        /src/client/status.go:260 +0x728
github.com/zrepl/zrepl/cli.(*Subcommand).run(0x121e440, 0xc000192780, 0x125d3b8, 0x0, 0x0)
        /src/cli/cli.go:105 +0xe2
github.com/spf13/cobra@v0.0.2/command.go:846 +0x2fe
        /go/pkg/mod/github.com/spf13/cobra@v0.0.2
github.com/zrepl/zrepl/cli.Run()
        /src/cli/cli.go:152 +0x2d
main.main()
        /src/main.go:24
```

Version info
```
# zrepl version
client: zrepl version=v0.3.1 go=go1.15.3 GOOS=linux GOARCH=amd64 Compiler=gc
server: zrepl version=v0.3.1 go=go1.15.3 GOOS=linux GOARCH=amd64 Compiler=gc

# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
```

Very, very impressed and delighted with `zrepl` overall! You're awesome, Christian, I hope you know that.

**Comment by problame** [comment]
> You're awesome, Christian, I hope you know that.
Thank you very much, I am flattered :blush: 

---

In general a crash in the status UI should not be able to affect the daemon.

---

## Analysis

https://github.com/zrepl/zrepl/blob/0a2dea05a952c6856934f6fec72f06df995c3377/client%2Fstatus.go#L120-L153

* yuck, `wrap()` is really not great code
* `width-INDENT_MULTIPLIER*t.indent` must be `-2`
* I think `t.indent` is 1 in the context where the crash happens
* INDENT_MULTIPLIER` is 4
* `width` must have been `2`. That's a really small terminal. But regardless, we should handle this correctly.

I'll hopefully fix this until the next release.

---

BTW: If you are interested you can try out the status-v2 UI, available at #297 , I'm pretty sure it doesn't have this bug.

**Comment by bazizsafi** [comment]
> `width` must have been 2. That's a really small terminal. But regardless, we should handle this correctly.

`zrepl status` is running in a `tmux` session in its own pane. My guess is that reconnecting to the `tmux` session in a too small terminal window resulted in this pane being squished to this non-sensical width.

> BTW: If you are interested you can try out the status-v2 UI

Nice! I'm looking forward to checking it out once [v0.3.2](https://github.com/zrepl/zrepl/milestone/11) is released and we updated our backup server.

Thanks for the fast response!

**Comment by problame** [comment]
We just merged #297 which removes the status v1 code. `wrap()` was not carried over to v2.
